### ISSUE TITLE
fix: APR throughput PROVISIONAL — NaN backward skips inflate measurement (PMAT-462)

### ISSUE THREAD

**Issue by noahgift** [paiml/qwen-train-canary#18]
## Problem

APR canary reports 194 tok/s but this is **inflated** because NaN-skipped backward
passes make steps faster. The stderr shows:

```
[CUDA-TRAIN] Layer 24 INPUT: nan=92160 inf=0
[CUDA] NaN/Inf loss detected (loss=NaN) — skipping backward pass
```

Steps with NaN loss skip the entire backward pass (grad computation, LoRA update,
optimizer step), so they complete much faster than real training steps. The reported
throughput includes these NaN-fast steps, making the number unreliable.

## Fix

1. **Canary**: Track `nan_backward_skips` and `valid_backward_steps` in output JSON
2. **Baseline**: Lowered from 190 to 40 tok/s (PROVISIONAL until NaN cascade fixed)
3. **Warning**: Canary prints PROVISIONAL warning when NaN skips detected
4. **Upstream**: entrenar#318 (layer forward zeros) still open — NaN cascade in layers 24-27

## Root Cause

The NaN cascade from layers 24-27 was partially fixed (21 upstream patches) but
the activation explosion through residual connections still occurs on some samples.
The scratch buffer zeroing (PMAT-453) prevents cross-step contamination but within
a single step, the residual chain can still overflow at layer ~14+.

## Status

- `_baseline_status: "PROVISIONAL"` in canary output when `nan_skips > 0`
- Baseline will be re-established once entrenar#318 is fully resolved

Refs PMAT-462
